### PR TITLE
add travis build for go versions 1.7.x -> 1.10.x

### PR DESCRIPTION
**Pull request by chemidy**
to ensure compatibility with older go version, add travis build from go 1.6.x to 1.10.x

### CODE REVIEW

**Comment by hiranya911** [comment]
Looks like some of our test dependencies don't work on older Go versions. If it's just the appengine package, we should remove that test dependency.

**Comment by chemidy** [comment]
> internal/http_client.go:47: req.WithContext undefined (type *http.Request has no field or method WithContext)

[WithContext](https://golang.org/doc/go1.7#net_http) is a go 1.7 feature

**Comment by chemidy** [comment]
Any idea for building issue with go get -t -v ./...  with go 1.6 and 1.7

`firebase.google.com/go/integration/auth
go build firebase.google.com/go/integration/auth: no buildable Go source files in 
/home/travis/gopath/src/firebase.google.com/go/integration/auth
firebase.google.com/go/integration/firestore
go build firebase.google.com/go/integration/firestore: no buildable Go source files in /home/travis/gopath/src/firebase.google.com/go/integration/firestore
firebase.google.com/go/integration/iid
go build firebase.google.com/go/integration/iid: no buildable Go source files in /home/travis/gopath/src/firebase.google.com/go/integration/iid
firebase.google.com/go/integration/messaging
go build firebase.google.com/go/integration/messaging: no buildable Go source files in /home/travis/gopath/src/firebase.google.com/go/integration/messaging
firebase.google.com/go/integration/storage
go build firebase.google.com/go/integration/storage: no buildable Go source files in /home/travis/gopath/src/firebase.google.com/go/integration/storage`

**Comment by chemidy** [comment]
`./firebase_test.go:465: t.Run undefined (type *testing.T has no field or method Run)
`

[t.Run](https://golang.org/doc/go1.7#testing) is a go 1.7 feature

File: internal/http_client.go [hunk 1]
```diff
@@ -37,14 +38,22 @@ type HTTPClient struct {
 	ErrParser ErrorParser
 }
 
+// Request contains all the parameters required to construct an outgoing HTTP request.
```
Comment (hiranya911) [inline internal/http_client.go]: Can we leave this where it used to be?

**Comment by chemidy** [comment]
Dropping golang 1.6 support is ok for me. 

I think the only sticking point is on appengine with legacy api_version: go1.6, but for now is already broken due to req.WithContext not supported.

**Comment by hiranya911** [comment]
@chemidy We are having an internal discussion on dropping golang 1.6 support. I will update this thread once a decision has been made.

**Comment by hiranya911** [comment]
Ok, we are officially ending support for Golang 1.6 and earlier (in all honesty we never supported them to begin with, as discovered through this PR). @chemidy can you clean this PR up so that Travis builds will run against Golang 1.7 and above? We can revert any changes you had to do to compile against 1.6.

**Comment by chemidy** [comment]
done

File: .travis.yml [hunk 2]
```diff
@@ -1,8 +1,30 @@
...
+install:
+    # prio to go 1.8 package with only test go get returns : "no buildable Go source files ..."
+    - go get -t -v $(go list ./... | grep -v integration)
+
 script:
     - golint -set_exit_status $(go list ./...)
-    - go test -v -test.short ./...
-
+    - go test -v -race -test.short ./...        # Run tests with the race detector.
+    - go vet -v ./...                           # Run Go static analyzer.
```
Comment (hiranya911) [inline .travis.yml]: Add newline at end of file.

File: .travis.yml [hunk 3]
```diff
@@ -1,8 +1,30 @@
...
+    - go: master
+  # Don't wait for tests to finish on allow_failures.
+  # Mark the build finished if tests pass on other versions of Go.
+  fast_finish: true
+
 go_import_path: firebase.google.com/go
+
 before_install:
-    - go get github.com/golang/lint/golint
+    - go get github.com/golang/lint/golint #Golint requires Go 1.6 or later.
```
Comment (hiranya911) [inline .travis.yml]: Add space after `#` --> `# Golint requires...`

File: .travis.yml [hunk 4]
```diff
@@ -1,8 +1,30 @@
...
+
+install:
+    # prio to go 1.8 package with only test go get returns : "no buildable Go source files ..."
```
Comment (hiranya911) [inline .travis.yml]: Lets change this as: `# Prior to golang 1.8, this can trigger an error for packages containing only tests.`

**Pull request merged**